### PR TITLE
chore: using common udsink across go routines.

### PR DESCRIPTION
**Pull request by yhl25**

### CODE REVIEW

File: pkg/sinks/udsink/sink.go [hunk 1]
```diff
@@ -18,21 +18,18 @@ package udsink
...
 	sinkpb "github.com/numaproj/numaflow-go/pkg/apis/proto/sink/v1"
-
 	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
 	"github.com/numaproj/numaflow/pkg/forward"
 	"github.com/numaproj/numaflow/pkg/forward/applier"
 	"github.com/numaproj/numaflow/pkg/isb"
 	"github.com/numaproj/numaflow/pkg/shared/logging"
 	"github.com/numaproj/numaflow/pkg/watermark/fetch"
 	"github.com/numaproj/numaflow/pkg/watermark/publish"
+	"go.uber.org/zap"
```
Comment (vigith) [inline pkg/sinks/udsink/sink.go:31]: fix importing

File: pkg/sinks/udsink/sink.go [hunk 2]
```diff
@@ -141,8 +126,3 @@ func (s *UserDefinedSink) Stop() {
 func (s *UserDefinedSink) ForceStop() {
 	s.isdf.ForceStop()
 }
-
-// IsHealthy checks if the udsink sidecar is healthy.
```
Comment (vigith) [inline pkg/sinks/udsink/sink.go:145]: why is this removed?

File: pkg/sinks/udsink/sink.go (hunk 1 above)
Comment (yhl25) [reply pkg/sinks/udsink/sink.go:31]: done

File: pkg/sinks/udsink/sink.go (hunk 2 above)
Comment (yhl25) [reply pkg/sinks/udsink/sink.go:145]: https://github.com/numaproj/numaflow/blob/cb1a38b4dc0ab9bf4ab4a789e3da59cf38564ea0/pkg/sinks/udsink/udsink_grpc.go#L49 we already have that method inside UDSgRPCBasedUDSink which will be used by the metrics server.

**Comment by vigith** [comment]
test is failing

**Comment by vigith** [comment]
for the time being you can skip the `Rater` test ref: #798

**Review by jy4096 — APPROVED** [review]

File: pkg/daemon/server/service/rater/rater_test.go [hunk 3]
```diff
@@ -107,8 +106,8 @@ func TestRater_Start(t *testing.T) {
 	}
 	r := NewRater(ctx, pipeline)
 	podTracker := NewPodTracker(ctx, pipeline, WithRefreshInterval(time.Second*5))
-	podTracker.httpClient = &mockHttpClient{podOneCount: 0, podTwoCount: 0, lock: &sync.RWMutex{}}
-	r.httpClient = &mockHttpClient{podOneCount: 0, podTwoCount: 0, lock: &sync.RWMutex{}}
+	//podTracker.httpClient = &mockHttpClient{podOneCount: 0, podTwoCount: 0, lock: &sync.RWMutex{}}
+	//r.httpClient = &mockHttpClient{podOneCount: 0, podTwoCount: 0, lock: &sync.RWMutex{}}
```
Comment (vigith) [inline pkg/daemon/server/service/rater/rater_test.go:110]: why do you have to comment these? isn't `t.SkipNow()` enough?

**Review by vigith — APPROVED** [review]

File: pkg/daemon/server/service/rater/rater_test.go (hunk 3 above)
Comment (yhl25) [reply pkg/daemon/server/service/rater/rater_test.go:110]: lint was failing
